include/swift/Sema/CSBindings.h

```diff
@@ -279,26 +279,26 @@ struct PotentialBindings {
   bool isPotentiallyIncomplete() const;
 
   /// If this type variable doesn't have any viable bindings, or
-  /// if there is only one binding and it's a hole type, consider
-  /// this type variable to be a hole in a constraint system
-  /// regardless of where hole type originated.
-  bool isHole() const {
-    if (isDirectHole())
+  /// if there is only one binding and it's a placeholder type, consider
```

Comment:
@Jumhyn @hborla What do you think about keeping "hole" terminology in the constraint system? My personal preference would be to keep calling "untyped" places in constraint system holes and call them placeholders in AST.

Reply:
I agree - I like the term "hole" for the constraint system because "placeholder" implies that it can be replaced, but if we bind to a placeholder in the constraint system, that means it solver could not replace it with a real type.

Reply:
Sounds good to me, I don't have super strong feelings either way! So we'd keep the type name as `PlaceholderType`, but all the constraint system methods (such as `isDirectHole()`) stay as is?

Reply:
Yes, that’s the idea.

Reply:
Effectively we are just utilizing a placeholder in a solution and AST to represent places which couldn’t be inferred.

Reply:
@hborla @xedin Thoughts on `TVO_CanBindToPlaceholder` vs. `TVO_CanBindToHole`? This is the only one in the constraint system that seems to me like the use of "hole" is perhaps more confusing than "placeholder", since it most literally represents whether we will bind the type var to a `PlaceholderType` or not. But I'm good with whatever y'all think is best!

Reply:
I think we should leave it as hole since it would be consistent with the rest.